pkg/network/dhcpconfigurator/dhcp_configurator.go

```diff
@@ -0,0 +1,79 @@
+package dhcpconfigurator
+
+import (
+	"fmt"
+	"os"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+)
+
+type DhcpConfigurator struct {
+	advertisingIfaceName     string
+	dhcpConfigurationFactory cache.InterfaceCacheFactory
+	filterByMac              bool
+	handler                  netdriver.NetworkHandler
+	launcherPID              string
+}
+
+// NewDhcpConfiguratorWithClientFilter should be used when the DHCP server is
+// expected to only reply to the MAC specified in the `cache.DhcpConfig` struct
+func NewDhcpConfiguratorWithClientFilter(configurationFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) DhcpConfigurator {
+	return DhcpConfigurator{
+		advertisingIfaceName:     advertisingIfaceName,
+		dhcpConfigurationFactory: configurationFactory,
+		launcherPID:              launcherPID,
+		filterByMac:              true,
+		handler:                  handler,
+	}
+}
+
+// NewDhcpConfigurator should be used when the DHCP server is
+// expected to reply all client requests, independently of their MAC address
+func NewDhcpConfigurator(configurationFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) DhcpConfigurator {
+	return DhcpConfigurator{
+		advertisingIfaceName:     advertisingIfaceName,
+		dhcpConfigurationFactory: configurationFactory,
+		launcherPID:              launcherPID,
+		filterByMac:              false,
+		handler:                  handler,
+	}
+}
+
+func (d DhcpConfigurator) ImportConfiguration(ifaceName string) (*cache.DhcpConfig, error) {
```

Comment:
The Import/Export is a little missleading, what do you thin of "ReadConfig"/"WriteConfig"

Reply:
Why misleading ? This exists literally so a process can import a dhcp configuration (or a domain interface, or ...) from another process.

Read / write conveys a lot less information than this.